pkg/sdk/testint/helpers_test.go

```diff
@@ -91,7 +91,7 @@ func createDatabaseWithOptions(t *testing.T, client *sdk.Client, id sdk.AccountO
 	return database, func() {
 		err := client.Databases.Drop(ctx, id, nil)
 		require.NoError(t, err)
-		err = client.Sessions.UseSchema(ctx, testSchema(t).ID())
+		err = testClient(t).Sessions.UseSchema(ctx, testSchema(t).ID())
```

Comment:
that's true, but you can also see that we force which schema to use. It will throw an error when creating a database on the other account where no "testSchema" exists (it's on the other account -> will throw "does not exist or not authorized error"). Maybe this shouldn't be here? I assume this is here because some tests needed that. I would explicitly call UseSchema in those places because I don't think there are that many of them. wdyt ?

Reply:
For tests with shares, I would leave creating new databases. Shares are database-related things where each test can cause different results in others. That being said I think I would go with 1 and 2.
1. have a function ```createDatabase(t *testing.T, client *sdk.Client, schema sdk.DatabaseObjectIdentifier)``` and two overrides: first one with implicit testClient and testSchema (testClientCreateDatabase() or something like that) and second one with implicit testSecondaryClient and testSecondarySchema
2. have similar objects created for the secondary client (objects and cleanup functions added to testint.integrationTestContext function)

Reply:
I did not suggest using the database/schema created during the setup in these tests, but creating them in setup would make this helper function pass (of course if we use the same name for both accounts, which I think is a valid idea to have an easier association between objects created for a given test run). Therefore, I think that we could solve it without introducing two overrides but I would not object strongly against it.

Reply:
As discussed created db, schema, and warehouse on the second account with the same names, so we can use TestDatabaseName and TestSchemaName (in createDatabase UseSchema) and it will work on either client.